R/estimate_uncertainty.R

```diff
@@ -72,27 +70,17 @@
 #'   point_nowcast_matrices = retro_nowcasts,
 #'   truncated_reporting_triangles = trunc_rts,
 #'   retro_reporting_triangles = retro_rts,
-#'   n = 2,
-#'   error_model = fit_distribution,
-#'   error_args = list(observation_model_name = "normal"),
-#'   ref_time_aggregator = zoo::rollmean,
-#'   ref_time_aggregator_args = list(k = 1, align = "right")
+#'   ref_time_aggregator = function(x) zoo::rollsum(x, k = 2, align = "right")
 #' )
 #' disp_params_agg
 estimate_uncertainty <- function(
     point_nowcast_matrices,
     truncated_reporting_triangles,
     retro_reporting_triangles,
     n = length(point_nowcast_matrices),
-    error_model = fit_distribution,
-    error_args = list(observation_model_name = "negative binomial"),
-    ref_time_aggregator = zoo::rollsum,
-    ref_time_aggregator_args = list(
-      k = 1,
-      align = "right"
-    ),
-    delay_aggregator = rowSums,
-    delay_aggregator_args = list(na.rm = TRUE)) {
+    error_model = function(x, mu) fit_nb(x, mu),
```

Comment:
@seabbs Ok I have added this back in (you cannot just use `fitdistrplus::fit_dist()` from my understanding because it doesn't accept a vector of expectiations and observations. If we want to use `mle_dist()` we can replace the current `fit_nb` with this. 

My follow-up question is do we want to provide additional helper functions for other distributions, which I believe will still need these custom functions to pass in vectors of observations and predictions (similar to what is in the current `fit_distribution` function but I will remove it?

Reply:
the plan as we discussed it is to keep a renamed exported and improved fit_nb only with the suggestion in the previous PR beign to handle the looping over the data in a wrapper function that people can use if they want to get different dispersions see the previous PR for that suggestion 

The suggestion for fitdistrpllus from the previous PR was to make this its own issue, to plan to write a suggests wrapper that handles the formatting like primarycensored does and in the same essential way if we decide to go that way and to not add the custom fit distributions because of all the technical debt it adds

Reply:
Ok will make the `fit_by_delay()` wrapper and open a new issue for supporting fitdistrplus